src/app/src/actions/mergeFacilities.js

```diff
@@ -0,0 +1,100 @@
+import { createAction } from 'redux-act';
+
+import csrfRequest from '../util/csrfRequest';
+
+import {
+    logErrorAndDispatchFailure,
+    makeGetFacilityByOARIdURL,
+} from '../util/util';
+
+export const startFetchMergeTargetFacility =
+    createAction('START_FETCH_MERGE_TARGET_FACILITY');
+export const failFetchMergeTargetFacility =
+    createAction('FAIL_FETCH_MERGE_TARGET_FACILITY');
+export const completeFetchMergeTargetFacility =
+    createAction('COMPLETE_FETCH_MERGE_TARGET_FACILITY');
+export const clearMergeTargetFacility = createAction('CLEAR_MERGE_TARGET_FACILITY');
+export const updateMergeTargetFacilityOARID =
+    createAction('UPDATE_MERGE_TARGET_FACILITY_OAR_ID');
+
+export function fetchMergeTargetFacility() {
+    return (dispatch, getState) => {
+        const {
+            mergeFacilities: {
+                targetFacility: {
+                    oarID,
+                },
+            },
+        } = getState();
+
+        if (!oarID) {
+            return null;
+        }
+
+        dispatch(startFetchMergeTargetFacility());
+
+        return csrfRequest
+            .get(makeGetFacilityByOARIdURL(oarID))
+            .then(({ data }) => dispatch(completeFetchMergeTargetFacility(data)))
+            .catch(err => dispatch(logErrorAndDispatchFailure(
+                err,
+                'An error prevented fetching that facility',
+                failFetchMergeTargetFacility,
+            )));
+    };
+}
+
+export const startFetchFacilityToMerge = createAction('START_FETCH_FACILITY_TO_MERGE');
+export const failFetchFacilityToMerge = createAction('FAIL_FETCH_FACILITY_TO_MERGE');
+export const completeFetchFacilityToMerge = createAction('COMPLETE_FETCH_FACILITY_TO_MERGE');
+export const clearFacilityToMerge = createAction('CLEAR_FACILITY_TO_MERGE');
+export const updateFacilityToMergeOARID = createAction('UPDATE_FACILITY_TO_MERGE_OAR_ID');
+
+export function fetchFacilityToMerge() {
+    return (dispatch, getState) => {
+        const {
+            mergeFacilities: {
+                facilityToMerge: {
+                    oarID,
+                },
+            },
+        } = getState();
+
+        if (!oarID) {
+            return null;
+        }
+
+        dispatch(startFetchFacilityToMerge());
+
+        return csrfRequest
+            .get(makeGetFacilityByOARIdURL(oarID))
```

Comment:
Consider preventing the same facility from being loaded as both the target and the "to merge." I updated the merge API implementation issue (#600) to make sure we block this on the server side, but it would be relatively easy to validate on the client, too.

<img width="1271" alt="Screen Shot 2019-06-19 at 2 21 01 PM" src="https://user-images.githubusercontent.com/17363/59802257-80cdc300-929d-11e9-8dd7-f3dd896f94c8.png">

Reply:
I added this in 4ea28dd5444b192a2409c1f0ff35fe779bf2d692 in a way that will propagate the error messages up to the UI:

![Screen Shot 2019-06-20 at 12 04 24 PM](https://user-images.githubusercontent.com/4165523/59863925-bdf48c80-9353-11e9-8055-e3e27b5cf128.png)
![Screen Shot 2019-06-20 at 12 04 36 PM](https://user-images.githubusercontent.com/4165523/59863926-bdf48c80-9353-11e9-9c97-3a670ff40933.png)